# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsIContentInlines.h
# Commit: 4f30d7afdf22
# Full Hash: 4f30d7afdf22c652481513564b60b37053c0948b
# Author: Atila Butkovits <abutkovits@mozilla.com>
# Date: 2025-05-08 04:32:48
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Revert "Bug 1932150 - Fix build bustage" for causing build bustages.
#   
#   This reverts commit 0681b181d6a8a48ac460a97dacb7fca91f9989c3.
#   
#   Revert "Bug 1932150 - Fix lint failure" for causing Build bustage.
# ==============================================================================

diff -r 1e64a48e3517 -r 4f30d7afdf22 dom/base/nsIContentInlines.h
--- a/dom/base/nsIContentInlines.h	Wed May 07 20:11:46 2025 +0300
+++ b/dom/base/nsIContentInlines.h	Wed May 07 20:37:37 2025 +0300
@@ -144,8 +144,9 @@
   return ::GetFlattenedTreeParentNode<nsINode::eForStyle>(this);
 }
 
-inline nsINode* nsINode::GetFlattenedTreeParentNodeForSelection() const {
-  return ::GetFlattenedTreeParentNode<nsINode::eForSelection>(this);
+inline nsIContent* nsINode::GetFlattenedTreeParentNodeForSelection() const {
+  nsINode* parent = ::GetFlattenedTreeParentNode<nsINode::eForSelection>(this);
+  return (parent && parent->IsContent()) ? parent->AsContent() : nullptr;
 }
 
 inline bool nsINode::NodeOrAncestorHasDirAuto() const {